src/communication/xcWebSocketSubscriber.js

```diff
@@ -0,0 +1,62 @@
+define(["rx"], function (Rx) {
+	"use strict"
+
+	var Subscriber = function (webSocket, configuration) {
+	    this.webSocket = webSocket;
+	    this.configuration = configuration;
+	    this.observableMsg = Rx.Observable.fromEvent(webSocket, 'message');
+	}
+
+    //	    this.subscribedTopic = []; TODOooooooooooooo
+	Subscriber.prototype.getEventToSend = function (componentName, stateMachineName) {
+	    var topic = this.configuration.getSubscriberTopic(componentName, stateMachineName)
+        var data = {
+            "Header": { "IncomingType": 0 },
+            "JsonMessage": JSON.stringify({ "Topic": { "Key": topic }})
+        };
+        return data;
+    }
+
+
+	Subscriber.prototype.subscribe = function (componentName, stateMachineName, subscriberListener) {
+	    var codes = this.configuration.getCodes(componentName, stateMachineName);
+	    var data = this.getEventToSend(componentName, stateMachineName);
+	    this.webSocket.send(convertToWebsocketInputFormat(data));
+	    this.observableMsg
+        .filter(function (e) {
```

Comment:
Not very efficient ! you are parsing JSON events twice
try to parse in a map function for example
Then filter and subscribe